R/github.R

```diff
@@ -79,20 +79,33 @@ github_pat <- function(quiet = TRUE) {
     pat <- Sys.getenv(env_var)
     if (nzchar(pat)) {
       if (!quiet) {
-        message("Using github PAT from envvar ", env_var)
+        message("Using github PAT from envvar ", env_var, ". ",
+                "Use `gitcreds::gitcreds_set()` if you want to use the git ",
+                "credential store instead")
       }
       return(pat)
     }
   }
 
+  pat <- tryCatch(
+    gitcreds_get()$password,
```

Comment:
It feels like the `host` should appear here, i.e. what if we're installing from a GitHub Enterprise deployment? I realize this implies that `github_pat()` needs to gain a `host` argument.

Maybe the git credential store is only supported / consulted when `host == "api.github.com"`?

I've verified that this PR can send my github.com PAT to a different host (and, obviously, the install fails).

Reply:
Yes, that is the proper way, but that would need a major refactoring of remotes, which is probably not worth doing.

remotes already sends the token to wrong host, so not change of behavior there. The only change is that we also check the credential store now.

Reply:
Yes, everything you say makes sense.